### PR TITLE
added isSelectable prop for Tile.types; hide checkbox for non-selectable-item

### PR DESCRIPTION
**Pull request by yumikohey**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Currently, in Tile View, if item is not selectable, Fabric shows a disabled "checkbox". Ideally, we should not show a checkbox for non-selectable item.

Before Changes:
checkbox shown up for non-selectable-item

#### Focus areas to test
Tile View

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10306)

### CODE REVIEW

**Comment by kenotron** [comment]
@ThomasMichon - please take a look

File: packages/experiments/src/components/Tile/Tile.tsx [hunk 1]
```diff
@@ -168,7 +169,7 @@ export class Tile extends React.Component<ITileProps, ITileState> {
 
     const { isSelected = false, isModal = false } = this.state;
 
-    const isSelectable = !!selection && selectionIndex > -1;
+    const isSelectable = isItemSelectable || (!!selection && selectionIndex > -1);
```
Comment (ThomasMichon) [inline packages/experiments/src/components/Tile/Tile.tsx]: This should be:
```ts
const {
  isSelectable = !!selection && selectionIndex > -1
} = this.props;
```
Basically, prefer the new value, falling back to the old value if `props.isSelectable` is `undefined`.

Comment (yumikohey) [reply packages/experiments/src/components/Tile/Tile.tsx]: @ThomasMichon  updated PR based on suggestion :)

File: packages/experiments/src/components/Tile/Tile.tsx [hunk 2]
```diff
@@ -207,7 +207,7 @@ export class Tile extends React.Component<ITileProps, ITileState> {
         href={href}
         onClick={onClick}
         ref={this.props.linkRef}
-        data-selection-invoke={isInvokable && selectionIndex > -1 ? true : undefined}
+        data-selection-invoke={isSelectable || (isInvokable && selectionIndex > -1) ? true : undefined}
```
Comment (ThomasMichon) [inline packages/experiments/src/components/Tile/Tile.tsx]: It's possible to select items but not be able to invoke them. We need `isInvokable` to be enforced, still.

**Pull request merged**